t/sharness.d/01-setup.sh

```diff
@@ -47,6 +47,8 @@ if test -n "$FLUX_TEST_INSTALLED_PATH"; then
     PATH=$FLUX_TEST_INSTALLED_PATH:$PATH
     fluxbin=$FLUX_TEST_INSTALLED_PATH/flux
 else # normal case, use ${top_builddir}/src/cmd/flux
+    PATH=$($FLUX_BUILD_DIR/src/cmd/flux python -c \
+           'import os,sys; print(os.path.dirname(sys.executable))'):$PATH
```

Comment:
perhaps note some of the corner cases I noted in #5091 that this does not fix?  Or perhaps just point to the issue?

Reply:
I'll do a little testing and push another commit here if it works out ok.

Reply:
Ok, force-pushed this PR branch with an added test in `sharness.d/01-setup.sh` to ensure `flux python` matches `python3`

Reply:
:+1:

Reply:
Hm, well maybe that didn't work in CI. I'll check it out.

Reply:
Ah, in one builder we have two versions of Python installed that are "compatible enough" to both work for `make check`. I'll revert this extra check in favor of a descriptive comment.

Reply:
Ok, force-pushed removal of that  failed experiment and added a verbose comment to 01-setup.sh.

Reply:
comment looks good to me!